onnxruntime/core/providers/openvino/qdq_transformations/qdq_stripping.cc

```diff
@@ -0,0 +1,699 @@
+      auto q_zero_point_dt = GetQDQDataType(&q_node);
+
+      // Can ignore if this Q is uint16 as it won't be consumed by any supported node
+      if (q_zero_point_dt != ONNX_NAMESPACE::TensorProto_DataType_UINT16 &&
+          !IsQDQSandwichedBetweenSoftmaxAndConvMatMulOps(&q_node)) {
+        // if it's unequal, then it's a conversion between quantized types.
+        AddNode(initializers_to_keep, src_graph, dst_graph, node_unit.GetNode());
+        return true;
+      }
+    }
+  }
+
+  // Keep int8 DQ/Qs in int16 -> int8
+  // Don't keep any QDQs in int8 -> int16
+  // Beginning of a converting QDQ pair. Check if previous Q is int8 and the following Q has a converting zp
+  if (node_unit.OpType() == "DequantizeLinear" && edges_exist) {
+    const Node& dq_node = node_unit.GetNode();
+    const Node& i_q_node = *dq_node.InputNodesBegin();
+    const Node& o_q_node = *dq_node.OutputNodesBegin();
+
+    if (i_q_node.OpType() == "QuantizeLinear" && o_q_node.OpType() == "QuantizeLinear") {
+      auto dq_zero_point_dt = GetQDQDataType(&dq_node);
+
+      if (dq_zero_point_dt != ONNX_NAMESPACE::TensorProto_DataType_UINT16 &&
+          IsConnectedQPresent(src_graph, dst_graph.Nodes(), &dq_node, dq_node.InputDefs()) &&
+          !IsQDQSandwichedBetweenSoftmaxAndConvMatMulOps(&dq_node)) {
+        AddNode(initializers_to_keep, src_graph, dst_graph, node_unit.GetNode());
+        return true;
+      }
+    }
+  }
+  return false;
+}
+// Handles adding a standalone node unit (i.e., one not wrapped with DQ/Q ops) to the dst graph.
+static void AddStandaloneNodeUnit(onnxruntime::Graph& dst_graph, const onnxruntime::GraphViewer& src_graph,
+                                  const NodeUnit& node_unit,
+                                  std::set<std::string>& initializers_to_keep,
+                                  const logging::Logger& /* logger */) {
+  assert(node_unit.UnitType() == NodeUnit::Type::SingleNode);
+
+  if (HandleDoubleQDQ(dst_graph, src_graph, node_unit, initializers_to_keep)) return;
+
+  auto add_identity_op = [&](bool duplicate_dq) {
+    std::array<NodeArg*, 1> input_args, output_args;
+
+    // Case to handle standalone duplicate DQs. Just redirect this arg to the original DQ instead and change the
+    // arg type to FLOAT as we're replacing it with Identity
+    if (duplicate_dq && GetQDQDataType(&node_unit.GetNode()) != ONNX_NAMESPACE::TensorProto_DataType_UINT16) {
+      std::string orig_dq_name = node_unit.Outputs()[0].node_arg.Name();  // ex: dql_output/duplicated
+      std::unique_ptr<ONNX_NAMESPACE::TypeProto> type_proto = ONNX_NAMESPACE::TypeProto::Create();
+      type_proto->copy_from(node_unit.Inputs()[0].node_arg.TypeAsProto());
+      type_proto->mutable_tensor_type()->set_elem_type(ONNX_NAMESPACE::TensorProto_DataType_FLOAT);
+      orig_dq_name.erase(orig_dq_name.find(DuplicateDQ), std::string::npos);  // ex: dql_output
```

Comment:
This seems to be relying on an implementation detail of the EnsureUniqueDQForNodeUnit optimizer where duplicate DQ nodes get the name `{original_dq_node_name}/duplicated`.  Is there a way to avoid this or make it more robust? Are you encountering models in which a duplicated DQ is a `NodeUnit::Type::SingleNode` instead of a `NodeUnit::Type::QDQGroup`?

Reply:
As of now, I can't think of a better way to detect and reverse duplicate DQs. And yes, some duplicate DQs occur as SingleNodes in the graph and that's why this is required.

Reply:
let's revisit this.